libraries/node-core-library/src/Executable.ts

```diff
@@ -579,25 +579,26 @@ export class Executable {
           });
         }
         childProcess.on('error', (error: Error) => {
-          errorThrown = true;
-          reject(error);
+          // Wait to call reject() until any output is collected
+          errorThrown = error;
         });
-        childProcess.on('exit', (code: number | null) => {
+        childProcess.on('close', (exitCode: number | null, signal: NodeJS.Signals | null) => {
           if (errorThrown) {
-            // We've already rejected the promise
-            return;
+            reject(errorThrown);
           }
-          if (code !== 0 && throwOnNonZeroExitCode) {
-            reject(new Error(`Process exited with code ${code}`));
+          if (exitCode !== 0 && throwOnNonZeroExitCode) {
+            reject(new Error(`Process exited with code ${exitCode}`));
+          } else if (signal) {
```

Comment:
`exitCode !== 0` is true when `signal` is truthy, since `null !== 0`. It's better to check `signal` before checking `exitCode`.

Reply:
fixed